File: DataBinding/PageAndRDLX/JsonDataSource/VB.NET/JsonDataSource/DataLayer.vb
Imports System
Imports System.Collections.Generic
Imports System.Net
Imports System.Net.Http
Imports System.Net.Http.Headers
Imports System.Text
Imports System.Text.Json

' Provides the data used in the sample.
Friend NotInheritable Class DataLayer
	Public Shared Function CreateData() As String
		Const sourceUrl As String = "http://localhost:10395/customers/GetJson"
		Dim responseText As String = Nothing

		Using httpClient = New HttpClient()
			httpClient.DefaultRequestHeaders.Authorization = AuthenticationHeaderValue.Parse("Basic " & Convert.ToBase64String(Encoding.[Default].GetBytes("admin:1")))
			httpClient.DefaultRequestHeaders.Accept.Add(New MediaTypeWithQualityHeaderValue("application/json"))
			Dim request = New HttpRequestMessage(HttpMethod.Get, sourceUrl)
			Dim response = httpClient.SendAsync(request).Result
			Dim json = response.Content.ReadAsStringAsync().Result
			Return json
		End Using
	End Function
End Class

File: DataBinding/PageAndRDLX/JsonDataSource/C#/JsonDataSource/DataLayer.cs
using System;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;

namespace ActiveReports.Samples.JsonDataSource
{
	// Provides the data used in the sample.
	internal sealed class DataLayer
	{
		public static string CreateData()
		{
			const string sourceUrl = @"http://localhost:30187/customers/GetJson";

			using (var httpClient = new HttpClient())
			{
				httpClient.DefaultRequestHeaders.Authorization 
					= AuthenticationHeaderValue.Parse("Basic " + Convert.ToBase64String(Encoding.Default.GetBytes("admin:1")));
				httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
				var request = new HttpRequestMessage(HttpMethod.Get, sourceUrl);
				var response = httpClient.SendAsync(request).Result;
				var json = response.Content.ReadAsStringAsync().Result;
				return json;
			}
		}
	}
}
